### ISSUE TITLE
API plugin: New DRF plugin with basic mgmt command functionality

### ISSUE THREAD

**Issue by techgique** [open-oni/open-oni#603]
- Create a new Django app / Open ONI plugin
  - Name "plugin_api" to match pattern of other ONI plugins
    - Code name "geomys antre" ;-)
  - How do we require the DRF package via pip from within an ONI plugin / Django app?
- Mount URLs at sub-URI similar to other ONI plugins: https://github.com/open-oni/open-oni/blob/dev/docs/customization/configuration.md#onisiteurlspy
- Build to handle a basic command that takes no args, e.g. `batches`
  - https://docs.djangoproject.com/en/3.2/ref/django-admin/#running-management-commands-from-your-code
  - Add tests - Nonexistent command, ???
    - https://docs.djangoproject.com/en/4.1/topics/testing/tools/#management-commands
  - Start documentation

**Comment by techgique** [comment]
`batches` command documented as deprecated, so will just make a root description endpoint to facilitate proof of concept with plugin using DRF

**Comment by jechols** [comment]
If building this as a plugin is too much of a nightmare (getting it to interact with DRF in particular), just toss it in core behind a flag that defaults to "off"

**Comment by techgique** [comment]
Yep, that was my thinking as well :+1:

**Comment by raybrarian** [comment]
This one is also a critical blocker for progress in NCA/Open ONI integration for the Scoop grant (functionality that we want that's tied to this is the loading and purging of batches through automation).

**Comment by jechols** [comment]
@techgique since you're out the first week of the next work cycle, I'm trying to come up with what I need right away so NCA isn't stalled too long. I'm not sure if we should have separate tickets for various management commands or what, but I'll put stuff here and you can do what you want with it.

NCA's first needs are the `load_batch` and `purge_batch` commands. If we can get the APIs for those two done ASAP, that would be really helpful. Even just a raw "proof of concept" API with no authentication or anything, just so we can get NCA's batch automation built and tested, with auth later.

There's also some questions we'll need to answer pretty quick. For instance, do we want some kind of "dropoff" location for `load_batch`? Or do we want the API to take an absolute path on disk? `purge_batch` should be easier, but still probably has questions I'm not sure about. The earlier I can get any information you have, the better.

**Comment by techgique** [comment]
Yeah, we can definitely scope a PR to focus on just those critical management commands without auth etc to not hold up NCA integration.

I was thinking we'd go with absolute path on disk for the argument passed to load_batch. That's how it behaves via the command line so I don't think we'd want to change that. For practical purposes of organization I think it makes sense to use a directory where batches to be loaded are located. Others' organization of their batches might differ and have reason for multiple locations though, so don't want to limit the command to only be able to handle batches if they are in one location.

**Comment by techgique** [comment]
Posting the outline of the API for this issue here for reference.

## Endpoint Explanation

These paths assume the following in Open ONI's `onisite/urls.py`:

```python
urlpatterns = [
    …
    path('api/admin/', include("onisite.plugins.admin_api.urls")),`
    …
]
```

- `api/admin/`
  - Requests:
    - GET
    - No JSON body or params
  - Response:
    - JSON object with `info` and `paths` keys. `info` object has title, summary, and version info. `paths` object has
      endpoint paths as keys and objects with more detailed info for each endpoint.
- `api/admin/batch/load`
  - Requests:
    - POST
      - Accepts JSON request body with `batch_path` key and string value of absolute path to batch directory
        (e.g. `{'batch_path': '/opt/openoni/data/batches/batch_nbu_manyissues_ver01'}`)
      - Responses:
        - 200 for attempting to load the batch with response:
          ```json
          {
            'info': 'Batch load attempted',
            'job_id': '(32-character hex UUID)',
            'status': '(In Progress|Succeeded|Failed)',
          }
          ```
        - 400 for invalid batch path with `{'info': 'Invalid batch path'}` response
        - 409 when another job with the same batch is in progress with response:
          ```json
          {
            'info': 'Job for (batch_name) already in progress',
            'job_id': '(32-character hex UUID)',
          }
          ```
        - 500 for server errors with `{'info': '(error info)'}` response
- `api/admin/batch/purge`
  - Requests:
    - POST
      - Accepts JSON request body with `batch_name` key and string value of batch name
        (e.g. `{'batch_name': 'batch_nbu_manyissues_ver01'}`)
      - Responses:
        - 200 for attempting to purge the batch with response:
          ```json
          {
            'info': 'Batch purge attempted',
            'job_id': '(32-character hex UUID)',
            'status': '(In Progress|Succeeded|Failed)',
          }
          ```
        - 400 for invalid batch name with `{'info': 'Invalid batch path'}` response
        - 409 when another job with the same batch is in progress with response:
          ```json
          {
            'info': 'Job for (batch_name) already in progress',
            'job_id': '(32-character hex UUID)',
          }
          ```
        - 500 for server errors with `{'info': '(error info)'}` response
- `api/admin/job/status`
  - Requests:
    - GET
      - Accepts JSON request body with `job_id` key and value to know which job to query
        (e.g. `{'job_id': '(32-character hex UUID)'}`)
      - Responses:
        - 200 for successful status check with `{'status': '(In Progress|Succeeded|Failed)'}` response
          - If the job status is "In Progress", additional key `page_count` will return # of pages in the batch
        - 400 for invalid job id with `{'info': 'Invalid job id'}` response
        - 500 for server errors with `{'info': '(error info)'}` response
